### PR TITLE
Use codeql.exe instead of codeql.cmd on Windows

### PR DESCRIPTION
**Pull request by nickrolfe**
Draft: seeing if it affects actions.

### CODE REVIEW

**Comment by aeisenberg** [comment]
Thanks for putting this up. This looks fine to me. What's the purpose of the `codeql.cmd` file? Looks like a thin wrapper around the exe. Is it just legacy?

**Comment by nickrolfe** [comment]
At this point it's just legacy.

In the first release of the CodeQL CLI, there was no `.exe` and all the launcher logic was in `codeql.cmd` (mirroring the shell script used on Linux and Mac), but this was difficult to maintain and didn't always work reliably for me in PowerShell, so we wrote a C++ launcher executable and made the `.cmd` version a thin wrapper around that.

I already tried making these changes with https://github.com/github/vscode-codeql/pull/211, but we had to revert that until we'd be sure all used had updated to a recent CLI that included the `.exe`.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -80,7 +80,7 @@ jobs:
         if: matrix.os == 'windows-latest'
         run: |
           cd extensions/ql-vscode
-          $env:CODEQL_PATH=$(Join-Path $env:GITHUB_WORKSPACE -ChildPath 'codeql-home/codeql/codeql.cmd')
+          $env:CODEQL_PATH=$(Join-Path $env:GITHUB_WORKSPACE -ChildPath 'codeql-home/codeql/exe.cmd')
```
Comment (aeisenberg) [inline .github/workflows/main.yml]: I think you mean `codeql.exe`.

**Comment by aeisenberg** [comment]
I cherry-picked the changes here into #284. Closing this one.

**Comment by aeisenberg** [comment]
The failing build appears to only affect 2.0.5. Now that 2.0.6 is out, the build is passing. Perhaps, it was a windows defender issue.

**Pull request closed without merging**